### DISCUSSION TITLE
The Good, the Bad and the Ugly!

### DISCUSSION THREAD

**Discussion by vekgithubb** [woocommerce/woocommerce#41975] [General]
Here we have a three star review. Three reasons why WooCommerce is good and three reasons why WooCommerce is bad.

**The Good**

1. It is true. WooCommerce is a good plugin that provides its features very generously and for free so that they can start their online business at almost no cost.
2. This plugin is used whether you want to sell physical products or you want to sell downloadable and digital products and it works better than any other popular plugin in this field.
3. A lot of possibilities is what is possible with WooCommerce. For example, you can easily add this feature to your WooCommerce store by installing a shopping cart reminder plugin. Or a list of favorites, or anything else similar to that found in most major stores around the world. But we should also keep in mind that some of these features can be included in the WooCommerce core itself so that there is no need to pay for them and there is no need for many external plugins because many of them are either discontinued. , or they have high prices, or they have a negative impact on the performance of WooCommerce, or they reduce efficiency and security. However, this is a positive feature that many features can be made available to people with WooCommerce, but it is better that some other new features are removed from the side markets and placed in WooCommerce itself with higher compatibility and for free. So that efficiency and security problems are not created and the cost is reduced.

**the Bad**

1. There is one truth about WooCommerce that cannot be denied or ignored and every WordPress developer and webmaster knows this. The truth is that overall WooCommerce is very slow and has a negative impact on the performance and speed of the site. In general, if you compare a store site built with WordPress + WooCommerce and a store site built with systems It has been created like Shopify, Magento, Open Card, PrestaShop. In any case, with any number of products in the store and with any type of look and any type of shared or dedicated hosting or virtual server or any other platform and conditions you consider; WooCommerce is definitely the loser and at the bottom of the list in this race. And surprisingly, for example, sites built with OpenCard are much faster. Even on a very cheap shared hosting, OpenCard and the other systems I mentioned are much faster than WooCommerce. And this is very sad.
2. The HPOS feature is a good feature, but it requires a few extra steps and a lot of precision to activate it. It is enough to make a little mistake or forget something and everything will be messed up. It would be better if only one click and one button were enough to activate this feature and there is no need to activate and deactivate the tick several times or to go and come back and do some additional tasks. Only one button and one option was enough, and you can put all the other items in it to do things automatically. As another suggestion in this field, it is better that this possibility is not a separate feature, but by default should not be a separate feature and should exist in the WooCommerce plugin itself. This is something that should have been the same from the beginning and I don’t know why it was not noticed from the beginning. It is natural that the data should be stored in separate tables. One of the problems of WooCommerce and many other plugins in WordPress and in general in WordPress is that everything is inside and inside and everything is mixed together and many things are not separate from each other and neat and tidy. It reduces the efficiency, speed, security and weight of WordPress.
3. Just like the HPOS feature that increases the speed of ordering and storing and searching for orders. WooCommerce needs another similar feature. Speeding up and increasing the efficiency of products and separating product tables and their specifications and features from posts and any other type of posts. Products are very slow, adding new variables is time consuming and not interesting. Product registration speed is slow, product search speed is slow. The type of product post type is similar to post type, normal blog posts, and there is no difference between them, and they are all stored in the same post tables, which are better to be separated.

Some of these features that are added as a new feature and a new possibility to WooCommerce, such as HPOS, should not be a feature in principle. Rather, these are things that, if we want to look at it in a principled way and with correct rules, they should have been like this from the beginning. And we should not have mentioned them as a new feature. Some things were not right from the beginning. Anyway, it’s nice to have everything working out. It is better to fix them.

After all, we know you are working hard to come up with something better. We know that our opinion is important to you and you care about it. Thank you again for your efforts. We hope to see new features and major improvements in WooCommerce as soon as possible. We hope you care about this review and our opinion. We hope to see a lighter, faster, more efficient, better and more interesting WooCommerce. I still believe that WooCommerce is the best store building system in the world; But if we don’t ignore the truth, some things in other systems are not only better than WooCommerce, but much better than WooCommerce, which I mentioned some of them. Hoping for more success for you.

**Comment by rpldom** [comment]
I definitely second this!

**Comment by hassannawaz2** [comment]
The Bad: Where WooCommerce Falls Short
Speed Issues

You’re right—WooCommerce is slower than Shopify or OpenCart. But: Pair it with lightweight hosting (like Kinsta) and a caching plugin (WP Rocket), and it’s much snappier.

Database Bloat

Mixing products/posts in WordPress tables is messy. Until WooCommerce separates them, use optimization plugins (e.g., Advanced Database Cleaner).

Plugin Overload

Too many plugins = security/performance risks. Example: Instead of 5 price-tag plugins, use one (like WooCommerce Composite Products).

Visit> [https://funnymonikers.com/](https://funnymonikers.com/)
